### PR TITLE
Updated cgol script, and added a version without built-in functions

### PR DESCRIPTION
**Pull request by NicEastvillage**
Made an updated version of Game of Life. Also added a version without the built-in function `count`, but a user defined version instead.

### CODE REVIEW

File: dreamcode/v3/cgol.cell [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// When Moore is used in a state's logic, it works as an array of states of the 
+// cells in the neighbourhood
+neighbourhood Moore {
+    (-1, 1), (0, 1), (1, 1),
+    (-1, 0),          (1, 0),
+    (-1, -1), (0, -1), (1, -1)
+}
+
+// Last is the difinition of states. The state has a name, a color (RGB), and some logic.
+// The logic is applied to each cell with this state.
```
Comment (falkecarlsen) [inline dreamcode/v3/cgol.cell]:
```suggestion
// The logic is applied to each cell with this state at each tick.
```
I think it's worth mentioning that this happens at each update cycle/tick

File: dreamcode/v3/cgol-without-builtin-functions.cell [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if (aliveNeighbours < 2 || 3 < aliveNeighbours) {
+        become Dead;
+    }
+}
+
+// Replacement for the built in function called count
+function count(neighbourhood nei, state z) int {
+	let counter = 0;
+	for (int i = 0; i < 8; i = i + 1) {
+		counter = counter + 1;
```
Comment (falkecarlsen) [inline dreamcode/v3/cgol-without-builtin-functions.cell]: What does this actually count? Right now it just counts counter up to eight through the for-loop.

Neither `nei` or `z` are used, too.

File: dreamcode/v3/cgol-without-builtin-functions.cell [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+neighbourhood Moore {
+    (-1, 1), (0, 1), (1, 1),
+    (-1, 0),          (1, 0),
+    (-1, -1), (0, -1), (1, -1)
+}
+
+// Last is the difinition of states. The state has a name, a color (RGB), and some logic.
+// The logic is applied to each cell with this state.
+state Dead (255, 255, 255) {
+	// The count function returns how many states in neighbourhood is of a given state.
```
Comment (falkecarlsen) [inline dreamcode/v3/cgol-without-builtin-functions.cell]: One indentation too much, it seems.

**Review by falkecarlsen — CHANGES_REQUESTED** [review]
Nice. Seems like CGOL sans builtin-functions is faulty though, also minor formatting comments.

**Review by jenrik — COMMENTED** [review]
I have no comments besides what @cogitantium has already mentioned

**Review by falkecarlsen — APPROVED** [review]

**Pull request merged**